### PR TITLE
RISC-V compatibility layer 20240307

### PR DESCRIPTION
**Pull request by bedroge**
Rerunning/rebuilding this now...

### CODE REVIEW

**Comment by boegel** [comment]
@bedroge Should we create a `riscv` branch for this?

File: ansible/playbooks/roles/compatibility_layer/defaults/main.yml [hunk 1]
```diff
@@ -16,27 +16,25 @@ gentoo_prefix_path: /cvmfs/{{ cvmfs_repository }}/versions/{{ eessi_version }}/c
...
+# March 7: latest commit (includes a RISC-V fix for GCC)
+gentoo_git_commit: 7e595dc3c3330522878452ea98906046c0d0f45b
 prefix_required_space: 15 GB
 prefix_user_defined_trusted_dirs:
   - "/cvmfs/{{ cvmfs_repository }}/host_injections/{{ eessi_version }}/compat/{{ eessi_host_os }}/{{ eessi_host_arch }}/lib"
 prefix_mask_packages: |
   # stick to GCC 10.x; using a too recent compiler in the compat layer complicates stuff in the software layer,
   # see for example https://github.com/EESSI/software-layer/issues/151
-  >=sys-devel/gcc-11
+  #>=sys-devel/gcc-11
```
Comment (boegel) [inline ansible/playbooks/roles/compatibility_layer/defaults/main.yml:27]: Add a comment to clarify why we don't mask `gcc-11` anymore?

File: ansible/playbooks/roles/compatibility_layer/defaults/main.yml [hunk 2]
```diff
@@ -16,27 +16,25 @@ gentoo_prefix_path: /cvmfs/{{ cvmfs_repository }}/versions/{{ eessi_version }}/c
...
   >=dev-libs/openssl-3
 prefix_unmask_packages: |
-  # unmask older GCC to make it installable
-  =sys-devel/gcc-9*
+  # unmask older OpenSSL to make it installable
+  =dev-libs/openssl-1.1.1w
 prefix_bootstrap_use_flags: |
   # make sure that gold linker is installed with binutils
-  sys-devel/binutils gold
+  # sys-devel/binutils gold
```
Comment (boegel) [inline ansible/playbooks/roles/compatibility_layer/defaults/main.yml:37]: mention why (gold does not support RISC-V?)

File: install_compatibility_layer.sh [hunk 3]
```diff
@@ -13,6 +13,13 @@ STORAGE=
 VERSION=2023.06
 VERBOSE=
 
+# RISC-V needs some different default values
+if [[ $(uname -m) = "riscv64" ]]; then
+  CONTAINER=docker://ghcr.io/eessi/bootstrap-prefix:debian-sid
+  REPOSITORY=riscv.eessi.io
+  VERSION=2024.03
```
Comment (boegel) [inline install_compatibility_layer.sh:20]: `20240307`

**Review by boegel — CHANGES_REQUESTED** [review]

File: install_compatibility_layer.sh (hunk 3 above)
Comment (bedroge) [reply install_compatibility_layer.sh]: Changed in https://github.com/EESSI/compatibility-layer/pull/202/commits/9463141c446c18735fcde9c303943f074e4293c5.

File: ansible/playbooks/roles/compatibility_layer/defaults/main.yml (hunk 2 above)
Comment (bedroge) [reply ansible/playbooks/roles/compatibility_layer/defaults/main.yml:39]: https://github.com/EESSI/compatibility-layer/pull/202/commits/4d5eb5b9476eb51c6c619fd189e749dd0107523a adds a comments.

File: ansible/playbooks/roles/compatibility_layer/defaults/main.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Defaults file for the compatibility layer role.
 ---
-eessi_version: "2023.06"
+eessi_version: "2024.03"
```
[the remark this answers is not in the document]
Comment (bedroge) [reply ansible/playbooks/roles/compatibility_layer/defaults/main.yml]: Done: https://github.com/EESSI/compatibility-layer/pull/202/commits/4d5eb5b9476eb51c6c619fd189e749dd0107523a

**Comment by bedroge** [comment]
> @bedroge Should we create a `riscv` branch for this?

Yes, that's a good idea.

**Comment by bedroge** [comment]
Using the configuration from this PR the compat layer built without issues, and it has been ingested to the `riscv.eessi.io` repo.

**Comment by boegel** [comment]
merged via #204

**Pull request merged**